frontend-react/src/utils/DateTimeUtils.ts

```diff
@@ -56,3 +58,47 @@ export function formatDateWithoutSeconds(d: string) {
     });
     return newDate.replace(/,/, "");
 }
+
+/*
+ * format: "Mon, 7/25/2022"
+ * WARNING: Intl.DateTimeFormat() can be slow if called in a loop!
+ * Rewrote to just use Date to save cpu
+ * */
+export const dateShortFormat = (d: Date) => {
+    const dayOfWeek =
+        ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"][d.getDay()] || "";
+    return (
+        `${dayOfWeek}, ` +
+        `${d.getMonth() + 1}/${d.getDate()}/${d.getFullYear()}`
+    );
+};
+
+/**
+ * Result string is like "12h 34m 05.678s"
+ * No duration returns ""
+ * @param dateNewer Date
+ * @param dateOlder Date
+ */
+export const durationFormatShort = (dateNewer: Date, dateOlder: Date): string => {
+    const msDiff = dateNewer.getTime() - dateOlder.getTime();
+    const hrs = Math.floor(msDiff / (60 * 60 * 1000)).toString();
+    const mins = Math.floor((msDiff / (60 * 1000)) % 60).toString();
+    // 0.1200001 -> '0.12`
+    const secs = parseFloat(((msDiff / 1000) % 60).toFixed(3)).toString();
```

Comment:
Maybe wait to do the `.toString()` conversion in the actual `if(){}` block? Then we don't have to do these string comparisons below.

Reply:
This is copy+pasted original code, so not necessary to change for this scope of work.